Feature/error handling

JonasBorgesLM opened the pull request:
Etapa 6/6 - Tratamento de erros

JonasBorgesLM committed:
feat: :goal_net: atualizacao do tratamento de erros

Melhoria do retorno de erros usando Apollo-server-errors e http-status-codes

JonasBorgesLM committed:
test: :test_tube: adicionando testes de falha

Criacao de mais testes. Incluindo testes de falha.

felipewaku commented on test/index.spec.ts:
```diff
@@ -72,4 +72,118 @@ describe('GraphQL API Tests', () => {
+          birthDate
+        }
+      }
+    `;
```
É melhor organizar um arquivo de teste por query, assim facilita configuração dos testes e deixa a estrutura mais organizada

felipewaku commented on src/errors/error-messages.ts:
```diff
@@ -0,0 +1,40 @@
+export class ErrorMessages {
+  static emailAlreadyExists() {
+    return new CustomError(StatusCodes.BAD_REQUEST, 'Email already exists', 'Ensure the email is unique');
+  }
```
Assim, nos projetos acabamos também fazendo esse port meio errado de erros http para graphql, o graphql na teoria mais flexivel: https://spec.graphql.org/October2021/#sec-Errors
ENtão não é necessário retornar o erro http e você pode criar um código de erro customizado

felipewaku commented on package.json:
```diff
@@ -31,11 +31,13 @@
   "dependencies": {
     "@apollo/server": "4.10.5",
     "@prisma/client": "5.17.0",
+    "apollo-server-errors": "3.3.1",
```
esse pacote é da versão 3 do apollo server, comoe stá utilizando o 4 deveria utilizar esse: https://www.apollographql.com/docs/apollo-server/data/errors

JonasBorgesLM committed:
refactor: :white_check_mark: melhorando a estrutura dos testes automatizados

JonasBorgesLM committed:
refactor: :recycle: retirando os erros http das mensagens de erro

JonasBorgesLM committed:
refactor: :recycle: atualizando as mensagens de erro

Retirei a dependencia apollo-server-errors e utilizei uma estrutura interna do apollo para construir as mensagens de erro
